src/services/repo/useRepo.ts

```diff
@@ -0,0 +1,90 @@
+import { useQuery } from '@tanstack/react-query'
+import { z } from 'zod'
+
+import Api from 'shared/api'
+import { NetworkErrorObject } from 'shared/api/helpers'
+
+const RepositorySchema = z.object({
+  private: z.boolean(),
+  uploadToken: z.string().nullable(),
+  defaultBranch: z.string().nullable(),
+  yaml: z.string().nullable(),
+  activated: z.boolean(),
+  oldestCommitAt: z.string().nullable(),
+  active: z.boolean(),
+})
+
+export const RepoSchema = z.object({
+  owner: z
+    .object({
+      isAdmin: z.boolean().nullable(),
+      isCurrentUserPartOfOrg: z.boolean(),
+      isCurrentUserActivated: z.boolean().nullable(),
+      repository: RepositorySchema,
+    })
+    .nullable(),
+})
+
+const query = `
+    query GetRepo($owner: String!, $repo: String!){
+      owner(username:$owner){
+        isAdmin
+        isCurrentUserPartOfOrg
+        isCurrentUserActivated
+        repository: repositoryDeprecated(name:$repo){
+          private
+          uploadToken
+          defaultBranch
+          yaml
+          activated
+          oldestCommitAt
+          active
+        }
+      }
+    }
+`
+
+export interface UseRepoArgs {
```

Comment:
Don't think this needs to be exported

Reply:
good point! Fixed in [e2bac12](https://github.com/codecov/gazebo/pull/2688/commits/e2bac126d6a254940ad10e6ec1b1b016777f7c66)